tests/test_sqlPreFlight.py

```diff
@@ -303,6 +303,40 @@ def testPreFlightSkyMapUnits(self):
         rows = list(rows)
         self.assertEqual(len(rows), 0)
 
+    def testPreFlightSpatialMatch(self):
+        """Test involving spacial match using join tables.
+
+        Note that realistic test needs a resonably-defined SkyPix and regions
+        in registry tables which is hard to implement in this simple test.
+        So we do not actually fill registry with any data and all queries will
+        return empty result, but this is still useful for coverage of the code
+        that generates query.
+        """
+        registry = self.registry
+
+        # dataset types
+        registry.makeRun(collection="test")
+        storageClass = StorageClass("testDataset")
+        registry.storageClasses.registerStorageClass(storageClass)
+
+        calexpType = DatasetType(name="CALEXP", dataUnits=("Camera", "Visit", "Sensor"),
+                                 storageClass=storageClass)
+        registry.registerDatasetType(calexpType)
+
+        coaddType = DatasetType(name="deepCoadd_calexp", dataUnits=("SkyMap", "Tract", "Patch",
+                                                                    "AbstractFilter"),
+                                storageClass=storageClass)
+        registry.registerDatasetType(coaddType)
+
+        # without data this should run OK but return empty set
+        originInfo = DatasetOriginInfoDef(defaultInputs=["test"], defaultOutput="")
```

Comment:
What is the string "test" here, is is something defined elsewhere or some new name? I assume there is a dataset named test located somewhere as this is an input, but it is not clear. If there is a commonly shared dataset, should the name for it be factored out somewhere as a class variable?

Reply:
"test" is the name of the collection, same thing as was created at line 318. I will make it clearer by using local variable for both.